src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -170,6 +174,11 @@ public int Seek(int char0)
                 }
                 while (block.End != index)
                 {
+                    if (_processed > _maxHeaderLength)
```

Comment:
Sounds good. Another option might be to throw directly and let the caller wrap the exception.